internal/provider/serverless_clusters_data_source.go

```diff
@@ -0,0 +1,309 @@
+											MarkdownDescription: "Message for GCP PrivateLink information.",
+											Computed:            true,
+											Attributes: map[string]schema.Attribute{
+												"service_attachment_name": schema.StringAttribute{
+													MarkdownDescription: "The target GCP service attachment name for private access.",
+													Computed:            true,
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+						"encryption_config": schema.SingleNestedAttribute{
+							MarkdownDescription: "The encryption settings for the cluster.",
+							Computed:            true,
+							Attributes: map[string]schema.Attribute{
+								"enhanced_encryption_enabled": schema.BoolAttribute{
+									MarkdownDescription: "Whether enhanced encryption is enabled.",
+									Computed:            true,
+								},
+							},
+						},
+						"high_availability_type": schema.StringAttribute{
+							MarkdownDescription: "The high availability type of the clusterV1beta1. ZONAL: High availability within a single zone. REGIONAL: High availability across multiple zones within a region",
+							Computed:            true,
+						},
+						"version": schema.StringAttribute{
+							MarkdownDescription: "The version of the cluster.",
+							Computed:            true,
+						},
+						"created_by": schema.StringAttribute{
+							MarkdownDescription: "The email of the creator of the cluster.",
+							Computed:            true,
+						},
+						"create_time": schema.StringAttribute{
+							MarkdownDescription: "The time the cluster was created.",
+							Computed:            true,
+						},
+						"update_time": schema.StringAttribute{
+							MarkdownDescription: "The time the cluster was last updated.",
+							Computed:            true,
+						},
+						"user_prefix": schema.StringAttribute{
+							MarkdownDescription: "The unique prefix in SQL user name.",
+							Computed:            true,
+						},
+						"state": schema.StringAttribute{
+							MarkdownDescription: "The state of the cluster.",
+							Computed:            true,
+						},
+						"labels": schema.MapAttribute{
+							MarkdownDescription: "The labels of the cluster.",
+							Computed:            true,
+							ElementType:         types.StringType,
+						},
+						"annotations": schema.MapAttribute{
+							MarkdownDescription: "The annotations of the cluster.",
+							Computed:            true,
+							ElementType:         types.StringType,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (d *serverlessClustersDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var data serverlessClustersDataSourceData
+	diags := req.Config.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Trace(ctx, "read serverless clusters data source")
+	clusters, err := d.provider.ServerlessClient.ListClusters(ctx)
```

Comment:
we need to list all clusters if we do not provide page/pagesize options.
@zhangyangyu do we need to provide the following options?
![image](https://github.com/user-attachments/assets/fbd0a707-e429-4ca7-b88b-86a3add29e55)